spec/services/proofing/resolution/progressive_proofer_spec.rb

```diff
@@ -198,101 +245,100 @@
-            instance_double(Proofing::Resolution::Result)
-          end
           before do
             allow(instance).to receive(:proof_id_address_with_lexis_nexis_if_needed).
-              and_return(result_that_failed_instant_verify)
-            allow(instant_verify_proofer).to receive(:proof).with(hash_including(state_id_address)).
-              and_return(result_that_failed_instant_verify)
+              and_return(instant_verify_proofer_result)
             allow(instance).to receive(:user_can_pass_after_state_id_check?).
-              with(result_that_failed_instant_verify).
+              with(instant_verify_proofer_result).
               and_return(true)
-            allow(result_that_failed_instant_verify).to receive(:success?).
+            allow(instant_verify_proofer_result).to receive(:success?).
               and_return(false)
           end
 
+          it 'includes the state ID in the InstantVerify call' do
+            proof
+
+            expect(instant_verify_proofer).to have_received(:proof).with(hash_including(state_id_address))
+          end
+
           context 'the failure can be covered by AAMVA' do
             before do
-              allow(result_that_failed_instant_verify).
+              allow(instant_verify_proofer_result).
                 to receive(:attributes_requiring_additional_verification).
                 and_return([:address])
             end
 
             context 'it is not covered by AAMVA' do
-              let(:failed_aamva_proof) { instance_double(Proofing::StateIdResult) }
+              let(:failed_aamva_proof) do
+                double = instance_double(Proofing::StateIdResult)
+                allow(double).to receive(:verified_attributes).and_return([])
+                allow(double).to receive(:success?).and_return(false)
+                double
+              end
+
               before do
                 allow(aamva_proofer).to receive(:proof).and_return(failed_aamva_proof)
-                allow(failed_aamva_proof).to receive(:verified_attributes).and_return([])
-                allow(failed_aamva_proof).to receive(:success?).and_return(false)
               end
-              it 'indicates the aamva check did not pass' do
-                result = subject
 
-                expect(result.state_id_result.success?).to eq(false)
+              it 'indicates the aamva check did not pass' do
+                expect(proof.state_id_result.success?).to eq(false)
               end
             end
 
             context 'it is covered by AAMVA' do
-              let(:successful_aamva_proof) { instance_double(Proofing::StateIdResult) }
+              let(:successful_aamva_proof) do
+                double = instance_double(Proofing::StateIdResult)
+                allow(double).to receive(:verified_attributes).
+                  and_return([:address])
+                allow(double).to receive(:success?).and_return(true)                
+                double
+              end
```

Comment:
another apporach is to make one "result" let value and then override properties
```suggestion
              let(:verified_attributes) { [:address] }
              let(:success) { true }
              let(:successful_aamva_proof) do
                instance_double(Proofing::StateIdResult, verified_attributes:, success?: success)
              end
```